Framework/Crystal/test/SCDCalibratePanels2Test.h

```diff
@@ -0,0 +1,877 @@
+//       is enabled for regression test.
+//       To test other testing targets, change the lead word from
+//                 run_ to test_
+//       to run them within the ctest framework.
+//       You might need to do one at a time to avoid ctest timeout error
+//       locally.
+
+// DEVNOTE:
+//  - cos, sin func uses radians
+//  - Quat class uses degrees
+
+#pragma once
+
+#include "MantidAPI/AlgorithmManager.h"
+#include "MantidAPI/AnalysisDataService.h"
+#include "MantidAPI/ExperimentInfo.h"
+#include "MantidAPI/Run.h"
+#include "MantidAPI/Sample.h"
+#include "MantidCrystal/SCDCalibratePanels2.h"
+#include "MantidDataObjects/EventWorkspace.h"
+#include "MantidDataObjects/Workspace2D.h"
+#include "MantidGeometry/Crystal/CrystalStructure.h"
+#include "MantidKernel/Logger.h"
+#include "MantidKernel/Unit.h"
+
+#include <boost/filesystem.hpp>
+#include <boost/math/constants/constants.hpp>
+#include <boost/math/special_functions/round.hpp>
+#include <cxxtest/TestSuite.h>
+#include <stdexcept>
+
+using namespace Mantid::API;
+using namespace Mantid::Crystal;
+using namespace Mantid::DataObjects;
+using namespace Mantid::Geometry;
+using namespace Mantid::Kernel;
+
+namespace {
+/// static logger
+Logger g_log("SCDCalibratePanels2Test");
+} // namespace
+
+class SCDCalibratePanels2Test : public CxxTest::TestSuite {
+public:
+  // This pair of boilerplate methods prevent the suite being created statically
+  // This means the constructor isn't called when running other tests
+  static SCDCalibratePanels2Test *createSuite() {
+    return new SCDCalibratePanels2Test();
+  }
+  static void destroySuite(SCDCalibratePanels2Test *suite) { delete suite; }
+
+  // ----------------- //
+  // ----- Setup ----- //
+  // ----------------- //
+  /**
+   * @brief Construct a new SCDCalibratePanels2Test object
+   *
+   */
+  SCDCalibratePanels2Test()
+      : wsname("wsSCDCalibratePanels2Test"),
+        pwsname("pwsSCDCalibratePanels2Test"),
+        tmppwsname("tmppwsSCDCalibratePanels2Test"), // fixed workspace name
+        bank_xtop("bank73/sixteenpack"),             //
+        bank_xcenter("bank12/sixteenpack"),          //
+        bank_xbottom("bank11/sixteenpack"),          //
+        bank_yright("bank59/sixteenpack"),           //
+        bank_yleft("bank58/sixteenpack"),            //
+        bank_ytop("bank88/sixteenpack"),             //
+        bank_ybottom("bank26/sixteenpack"),          //
+        silicon_a(5.431), silicon_b(5.431), silicon_c(5.431),   // angstrom
+        silicon_alpha(90), silicon_beta(90), silicon_gamma(90), // degree
+        silicon_cs(CrystalStructure("5.431 5.431 5.431", "F d -3 m",
+                                    "Si 0 0 0 1.0 0.02")),
+        dspacing_min(1.0), dspacing_max(10.0),   //
+        wavelength_min(0.1), wavelength_max(10), //
+        omega_step(3.0),                         //
+        TOLERANCE_L(5e-4), // this calibration has intrinsic accuracy limit of
```

Comment:
Trying to push the limits as much as we could so that the future developers have a better understanding of the inherent accuracy limits of this algorithm.